src/test/java/apitest/TestByteSerializer.java

```diff
@@ -8,15 +10,16 @@
 import impl.gen.packet.LoginBody;
 import org.junit.Assert;
 import org.junit.Test;
+import service.ByteBufferService;
 import util.ByteUtil;
 import util.SocketChannelUtil;
 
-import java.nio.ByteBuffer;
-
 import static org.mockito.Mockito.spy;
 
 public class TestByteSerializer {
```

Comment:
I think this test does not test your functionality well, because it single threaded.

I propose to create new multi-threaded integration test without mocks to verify thread safeness of your code.
Let say you have single calculation server and 10 clients with stack of 10 integers in each.
Server responsibilities:
 - accept new connection
 - get client operation (one of +, -, *, hex), save it for corresponding client
 - wait for client inputs (while clients sending data)
 - for each request server has to response to the client that he received data and repeat data that was received.
 - when client requests result, sever calculates result of the data that it received and send it to the client.

Client responsibilities:
 - randomly choose and operation (one of +, -, *, hex), generate 10 random integers in interval (-100, 100)
 - connect to the server
 - send operation to the server
 - for each integer send it to the server and verify that server took it and response with the same value
 - ask for the result
 - verify calculation result

Test flow:
 - server start up
 - one client connects to the server and starts calculating
 - 1 second pause
 - new two clients connect to the server in parallel and start calculating
 - 1 second pause
 - new three clients connect to the server in parallel and start calculating
 - 1 second pause
 - last four clients connect to the server in parallel and start calculating
 
**Note:** operation hex - mean convert integer to hexadecimal string. server result is an array of strings.

you should use your code from this PR to simplify work.
When test will be ready and it pass you will get what do you need